streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionResolverJoinProcessorSupplier.java

```diff
@@ -0,0 +1,85 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals.foreignkeyjoin;
+
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Murmur3;
+import org.apache.kafka.streams.kstream.ValueJoiner;
+import org.apache.kafka.streams.kstream.internals.KTableValueGetter;
+import org.apache.kafka.streams.kstream.internals.KTableValueGetterSupplier;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+public class SubscriptionResolverJoinProcessorSupplier<K, V, VO, VR> implements ProcessorSupplier<K, SubscriptionResponseWrapper<VO>> {
+    private final KTableValueGetterSupplier<K, V> valueGetterSupplier;
+    private final Serializer<V> valueSerializer;
+    private final ValueJoiner<V, VO, VR> joiner;
+
+    public SubscriptionResolverJoinProcessorSupplier(final KTableValueGetterSupplier<K, V> valueGetterSupplier,
+                                                     final Serializer<V> valueSerializer,
+                                                     final ValueJoiner<V, VO, VR> joiner) {
+        this.valueGetterSupplier = valueGetterSupplier;
+        this.valueSerializer = valueSerializer;
+        this.joiner = joiner;
+    }
+
+    @Override
+    public Processor<K, SubscriptionResponseWrapper<VO>> get() {
+        return new AbstractProcessor<K, SubscriptionResponseWrapper<VO>>() {
+
+            private KTableValueGetter<K,V> valueGetter;
+
+            @Override
+            public void init(final ProcessorContext context) {
+                super.init(context);
+                valueGetter = valueGetterSupplier.get();
+                valueGetter.init(context);
+            }
+
+            @Override
+            public void process(K key, SubscriptionResponseWrapper<VO> value) {
+                final V currentValue = valueGetter.get(key).value();
+                //We are unable to access the actual source topic name for the valueSerializer at runtime, without
+                //tightly coupling to KTableRepartitionProcessorSupplier.
+                //While we can use the source topic from where the events came from, we shouldn't serialize against it
+                //as it causes problems with the confluent schema registry, which requires each topic have only a single
+                //registered schema.
+                String dummySerializationTopic = context().topic() + "-join-resolver";
+                long[] currentHash = (currentValue == null ?
```

Comment:
While it could go into a static method, it will always need a topic name unique between all topologies because of the weak interface definition of `serialization`. It is ambiguous as to whether topic can be null in `serialize(topic,data)`. From experience, the Confluent Avro Serde _will_ register once with a null topic, but then will forever give already-registered exceptions for all remaining registrations. This means that we will forever need some unique dummy topic per foreignKeyJoin schema, and it will need to live in some common place. 

I can create a dummy topic name in the `KTableImpl.doForeignKeyJoin()` and pass in the necessary common reference to both `SubscriptionResolverJoinProcessorSupplier` and `KTableRepartitionerProcessorSupplier`. This will ensure that `serialize(topic,data)` is consistent, since the topic will not rely on the topology (outside of uniqueness). 

I am very frustrated with the mandatory inclusion of "topic" in the serialization interface, combined with the ambiguously weak definition of if it can be null. It puts this code squarely in the "damned if you do, damned if you don't" category. 

In this example I do not NEED a topic whatsoever. In fact, I would prefer to just call `serialize(null, data)`. This, however, will end up throwing an exception if using the Confluent Avro Serde, since sometime, somewhere, someone will have registered a schema on `null` topic. While it is easy to say "don't worry about the Avro schema registry", there have already been a number of protests to the lack of supporting it. Therefore, I need some dummy value